misc/pcirebind/src/main.go

```diff
@@ -0,0 +1,154 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"strings"
+)
+
+type rebind struct {
+	id        string
+	oldDriver string
+	newDriver string
+}
+
+// writeToSysFile writes to the specified sysfs file
+func writeToSysFile(path, content string) error {
+	file, err := os.OpenFile(path, os.O_WRONLY, 0200) // 0200 is write-only permission
+	if err != nil {
+		e := fmt.Errorf("failed to open %s: %v", path, err)
+		fmt.Printf("[error] %v", e)
+
+		return e
+	}
+	defer file.Close()
+
+	if _, err = file.WriteString(content); err != nil {
+		e := fmt.Errorf("failed to write to %s: %v", path, err)
+		fmt.Printf("[error] %v", e)
+
+		return e
+	}
+
+	return nil
+}
+
+// parseKernelCmdline parses specially formatted strings out of /proc/cmdline
+//
+// The format is as follows:
+// pcirebind.rebind=<pci_bus_id>_<current_driver>+<new_driver>
+//
+// example:
+//
+//	pcirebind.rebind=0000:04:00.0_ixgbe+vfio-pci
+func parseKernelCmdline(readFunc func(string) ([]byte, error)) ([]rebind, error) {
```

Comment:
this could probably use `go-procfs` from here: https://github.com/siderolabs/go-procfs

Reply:
@smira i wonder if this needs to be an extension or core talos? :thinking: